lib/tesla/adapter/gun.ex

```diff
@@ -345,6 +366,18 @@ if Code.ensure_loaded?(:gun) do
 
     defp tunnel_tls_opts(opts, _, _), do: opts
 
+    defp add_proxy_auth_credentials(opts, %{proxy_auth: {username, password}})
+         when not is_nil(username) and not is_nil(password),
```

Comment:
maybe `is_binary/1` guard will be more suitable here for checking username and password?

Reply:
You are absolutely right, corrected.